receiver/mongodbatlasreceiver/mongodb_event_to_logdata.go

```diff
@@ -0,0 +1,166 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"encoding/json"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+)
+
+const (
+	// Number of log attributes to add to the plog.LogRecordSlice.
+	totalLogAttributes = 5
+
+	// Number of resource attributes to add to the plog.ResourceLogs.
+	totalResourceAttributes = 4
+)
+
+// layout for the timestamp format in the plog.Logs structure
+const layout = "2006-01-02T15:04:05.000-07:00"
+
+// Severity mapping of the mongodb atlas logs
+var severityMap = map[string]plog.SeverityNumber{
+	"F":  plog.SeverityNumberFATAL,
+	"E":  plog.SeverityNumberERROR,
+	"W":  plog.SeverityNumberWARN,
+	"I":  plog.SeverityNumberINFO,
+	"D":  plog.SeverityNumberDEBUG,
+	"D1": plog.SeverityNumberDEBUG,
+	"D2": plog.SeverityNumberDEBUG2,
+	"D3": plog.SeverityNumberDEBUG3,
+	"D4": plog.SeverityNumberDEBUG4,
+	"D5": plog.SeverityNumberDEBUG4,
+}
+
+// mongoAuditEventToLogRecord converts model.AuditLog event to plog.LogRecordSlice and adds the resource attributes.
+func mongodbAuditEventToLogData(logger *zap.Logger, e model.AuditLog, r resourceInfo) plog.Logs {
+	ld := plog.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	lr := sl.LogRecords().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	// Attributes related to the object causing the event.
+	resourceAttrs.InsertString("org", r.Org)
+	resourceAttrs.InsertString("project", r.Project.Name)
+	resourceAttrs.InsertString("cluster", r.Cluster.Name)
+	resourceAttrs.InsertString("hostname", r.Hostname)
```

Comment:
These attributes need to be named according to the semantic conventions where possible, and minimally according to the naming guidelines. Anything specific to mongo needs to have a mongo-specific namespace. Is one established for metrics already?

Reply:
Sure thing, I understand what you mean, but could you send the link for the semantic conventions and what not. I'm having a hard time finding them

Reply:
Yes btw there is one established for metrics
```
rm.Resource().Attributes().UpsertString("mongodb_atlas.host.name", val)
```

Reply:
Here are the resource semantic conventions: https://github.com/open-telemetry/opentelemetry-specification/tree/main/semantic_conventions/resource

Here are the naming guidelines for semantic conventions: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md